docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/providing-dynamic-dropdown-values/resources/liferay-z4h3.zip/z4h3-impl/src/main/java/com/acme/z4h3/internal/configuration/admin/definition/Z4H3ConfigurationFieldOptionsProvider.java

```diff
@@ -0,0 +1,63 @@
+package com.acme.z4h3.internal.configuration.admin.definition;
+
+import com.liferay.configuration.admin.definition.ConfigurationFieldOptionsProvider;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.osgi.service.component.annotations.Component;
+
+@Component(
+	property = {
+		"configuration.field.name=dynamicallyPopulatedColorTypes",
+		"configuration.pid=com.acme.z4h3.internal.configuration.Z4H3Configuration"
+	},
+	service = ConfigurationFieldOptionsProvider.class
+)
+public class Z4H3ConfigurationFieldOptionsProvider
+	implements ConfigurationFieldOptionsProvider {
+
+	public List<Option> getOptions() {
+		Stream<Color> stream = _colors.stream();
+
+		return stream.filter(
+			color -> color.label != null
+		).map(
+			color -> new Option() {
+
+				@Override
+				public String getLabel(Locale locale) {
+					return color.label;
+				}
+
+				@Override
+				public String getValue() {
+					return color.value;
+				}
+
+			}
+		).collect(
+			Collectors.toList()
+		);
```

Comment:
This code will work, but IMO I think the usage of `stream` here will probably distract from the core lesson. It will probably be a little clearer to just do a straightforward `for` loop and populate a list.